cmd/gctsCreateRepository.go

```diff
@@ -0,0 +1,156 @@
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	type repoData struct {
+		RID             string `json:"rid"`
+		Name            string `json:"name"`
+		Role            string `json:"role"`
+		Type            string `json:"type"`
+		VSID            string `json:"vsid"`
+		GithubURLstring string `json:"url"`
+	}
+
+	type createRequestBody struct {
+		Repository string   `json:"repository"`
+		Data       repoData `json:"data"`
+	}
+
+	type repoConfig struct {
+		Key      string `json:"key"`
+		Value    string `json:"value"`
+		Category string `json:"category"`
+	}
+
+	type createResultBody struct {
+		RID         string       `json:"rid"`
+		Name        string       `json:"name"`
+		Role        string       `json:"role"`
+		Type        string       `json:"type"`
+		VSID        string       `json:"vsid"`
+		Status      string       `json:"status"`
+		Branch      string       `json:"branch"`
+		URL         string       `json:"url"`
+		CreatedBy   string       `json:"createdBy"`
+		CreatedDate string       `json:"createdDate"`
+		Connection  string       `json:"connection"`
+		Config      []repoConfig `json:"config"`
+	}
+
+	type createResponseBody struct {
+		Repository createResultBody `json:"repository"`
+		Exception  string           `json:"exception"`
+	}
+
+	reqBody := createRequestBody{
+		Repository: config.Repository,
+		Data: repoData{
+			RID:             config.Repository,
+			Name:            config.Repository,
+			Role:            config.Role,
+			Type:            config.Type,
+			VSID:            config.VSID,
+			GithubURLstring: config.RemoteRepositoryURL,
+		},
+	}
+	jsonBody, marshalErr := json.Marshal(reqBody)
+
+	if marshalErr != nil {
+		return fmt.Errorf("creating the repository locally failed: %w", marshalErr)
+	}
+
+	header := make(http.Header)
+	header.Set("Content-Type", "application/json")
+	header.Add("Accept", "application/json")
+
+	url := "http://" + config.Host +
+		"/sap/bc/cts_abapvcs/repository?sap-client=" + config.Client
+
+	resp, httpErr := httpClient.SendRequest("POST", url, bytes.NewBuffer(jsonBody), header, nil)
+
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if resp == nil {
+		return fmt.Errorf("creating the repository locally failed: %w", httpErr)
+	}
+
+	var response createResponseBody
+	parsingErr := parseHTTPResponseBodyJSON(resp, &response)
+
+	if parsingErr != nil {
+		log.Entry().Warning(parsingErr)
+	}
+
+	if httpErr != nil {
+		if resp.StatusCode == 500 && response.Exception == "Repository already exists" {
+			log.Entry().
+				WithField("repository", config.Repository).
+				Info("the repository already exists locally")
+			return nil
+		}
+		return fmt.Errorf("creating the repository locally failed: %w", httpErr)
+	}
+
+	log.Entry().
+		WithField("repository", config.Repository).
+		Info("successfully created the local repository")
```

Comment:
The term "local" is misleading IMO. The repo is created in the ABAP-System, isn't it. That is not what I understand as "local". "local" would be on the CI/CD server IMO.